syntax = "proto2";

package PluginProtocolProto;

message PluginAPIMessage
{
    optional string pluginName = 1;
    enum CommandOption
    {
        SendEvent = 1;
        SendStatus = 2;
        Registration = 3;
        ApplyPolicy = 4;
        DoAction = 5;
        RequestStatus = 6;
        Telemetry = 7;
        RequestCurrentPolicy = 8;
        InvalidCommand = 9;
        RequestHealth = 10;
        SendThreatHealth = 11;
    }
    optional CommandOption command = 2;
    optional string applicationId = 3;
    repeated string payload = 4;
    optional string error = 5;
    optional bool acknowledge = 6;
    optional string correlationId = 7;
}

